packages/core/src/standards/http.ts

```diff
@@ -370,25 +374,54 @@ export function _isBodyStream(stream: ReadableStream): boolean {
 export type RequestInfo = BaseRequestInfo | Request;
 
 export interface RequestInit extends BaseRequestInit {
-  readonly cf?: IncomingRequestCfProperties | RequestInitCfProperties;
+  cf?: IncomingRequestCfProperties | RequestInitCfProperties;
 }
 
-type Mutable<T> = {
-  -readonly [P in keyof T]: T[P];
-};
-
 const enumerableRequestKeys: (keyof Request)[] = [
   "cf",
   "signal",
   "redirect",
   "url",
   "method",
 ];
+
+// These keys aren't marked enumerable on `BaseRequest`, so we can't use
+// `{...req}` to clone the init object. Instead, we use `in` checks and copy
+// them over if present.
+const requestInitKeys: (keyof RequestInit)[] = [
+  "method",
+  "keepalive",
+  "headers",
+  "body",
+  "redirect",
+  "integrity",
+  "signal",
+  "credentials",
+  "mode",
+  "referrer",
+  "referrerPolicy",
+  "window",
+  "dispatcher",
+  "duplex",
+];
+function cloneRequestInit(init: RequestInit): RequestInit {
+  const result: RequestInit = {};
+  for (const key of requestInitKeys) {
+    if (key in init) (result as Record<string, unknown>)[key] = init[key];
+  }
+  return result;
+}
+
 export class Request extends Body<BaseRequest> {
   // noinspection TypeScriptFieldCanBeMadeReadonly
   #cf?: IncomingRequestCfProperties | RequestInitCfProperties;
 
   constructor(input: RequestInfo, init?: RequestInit) {
+    // If body is a FixedLengthStream, set Content-Length to its expected length
+    const contentLength: number | undefined = (init?.body as any)?.[
```

Comment:
Can this be typed as something other than `any`? (I suspect not, given it seems like `kContentLength` is internal, but worth checking)